packages/cli/src/commands/git/connect.ts

```diff
@@ -101,10 +121,194 @@ export default async function connect(
+
+  const checkAndConnect = await checkExistsAndConnect(
+    client,
+    output,
+    confirm,
+    org,
+    project,
+    gitProviderLink,
+    provider,
+    repoPath,
+    parsedOrg,
+    repo
+  );
+  if (typeof checkAndConnect === 'number') {
+    return checkAndConnect;
+  }
+
+  output.log(
+    `Connected ${formatProvider(provider)} repository ${chalk.cyan(repoPath)}!`
+  );
+
+  return 0;
+}
+
+async function connectArg(
+  client: Client,
+  output: Output,
+  confirm: boolean,
+  org: Org,
+  project: Project,
+  repoArg: string
+) {
+  output.log(`Connecting Git remote: ${link(repoArg)}`);
+  const parsedRepoArg = parseRepoUrl(repoArg);
+  if (!parsedRepoArg) {
+    output.error(
+      `Failed to parse URL "${repoArg}". Please ensure the URL is valid.`
+    );
+    return 1;
+  }
+  const { provider, org: parsedOrg, repo } = parsedRepoArg;
+  const repoPath = `${parsedOrg}/${repo}`;
+  const connect = await checkExistsAndConnect(
+    client,
+    output,
+    confirm,
+    org,
+    project,
+    project.link,
+    provider,
+    repoPath,
+    parsedOrg,
+    repo
+  );
+  if (typeof connect === 'number') {
+    return connect;
+  }
+  output.log(
+    `Connected ${formatProvider(provider)} repository ${chalk.cyan(repoPath)}!`
+  );
+  return 0;
+}
+
+async function connectArgWithLocalGit(
+  client: Client,
+  output: Output,
+  org: Org,
+  project: Project,
+  confirm: boolean,
+  gitConfig: Dictionary<any>,
+  repoArg: string
+) {
+  const parsedUrlArg = parseRepoUrl(repoArg);
+  if (!parsedUrlArg) {
+    output.error(
+      `Failed to parse URL "${repoArg}". Please ensure the URL is valid.`
+    );
+    return 1;
+  }
+
+  const remoteUrls = pluckRemoteUrls(gitConfig);
+  if (remoteUrls) {
+    const shouldConnect = await promptConnectArg(
+      client,
+      output,
+      confirm,
+      repoArg,
+      remoteUrls
+    );
+    if (!shouldConnect) {
+      return 1;
+    }
+    if (shouldConnect) {
+      const { provider, org: parsedOrg, repo } = parsedUrlArg;
+      const repoPath = `${parsedOrg}/${repo}`;
+      output.log(`Connecting Git remote: ${link(repoArg)}`);
+      const connect = await checkExistsAndConnect(
+        client,
+        output,
+        confirm,
+        org,
+        project,
+        project.link,
+        provider,
+        repoPath,
+        parsedOrg,
+        repo
+      );
+      if (typeof connect === 'number') {
+        return connect;
+      }
+      output.log(
+        `Connected ${formatProvider(provider)} repository ${chalk.cyan(
+          repoPath
+        )}!`
+      );
+    }
+    return 0;
+  }
+  return await connectArg(client, output, confirm, org, project, repoArg);
+}
+
+async function promptConnectArg(
+  client: Client,
+  output: Output,
+  yes: boolean,
+  arg: string,
+  remoteUrls: Dictionary<string>
+) {
+  const multiple = Object.keys(remoteUrls).length > 1;
+  if (multiple) {
+    output.log(`Found multiple Git repositories in your local Git config:`);
```

Comment:
Do we encourage template literals if there are no expressions? I've seen this in a few other places in the PR.

Reply:
I don't think there's a standard, but I agree that it's probably better to not use template literals here.